### ISSUE TITLE
AGN power law model failing to generate plausible spectrum with certain inputs

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#229]
In an attempt to duplicate the results in the Mehdipour+16 paper, I tried to use the pure power law model with limits set from 0.1 eV to 1e6 eV.  These fail to produce a plausible spectrum and ultimately result in segmentation faults in the program.   An example .pf file is attached.

[power_34.pf.txt](https://github.com/agnwinds/python/files/685043/power_34.pf.txt)

This needs to be fixed both for the general user and for our comparisons with other routines.

**Comment by Higginbottom** [comment]
The first problem I've found is that the AGN type in the pf file is set to power law, but the photon sampling approach is set to cloudy table. the cloudy table photon sampling uses the high and low energy breaks, which are unset because of the AGN type. this causes problems. we should check for this mismatch. 
However,  fixing this in the pf file still results in a segfault even though the banding looks to be ok. Investigating

**Comment by Higginbottom** [comment]
OK, i thing the new segfault was just coz i hadn't recompiled properly. It seems to be OK as long as the modes match. I do get bad photon errors with alpha=-1.0, probably because of the integration issues when we get a div by zero. This should also be fixed!

**Comment by Higginbottom** [comment]
Right, this is all fixed.
With regard to the 'user error' that started this off, I've simply added a check. If you ask for a standard power law for the AGN spec type, then ask for a broken power law banding, it throws an error and quits. 
Then regarding the second error, that is the alpha=-1 problem which causes a crash, I've just recoded so the integrals in the power law parts of the code check to see if alpha=-1 and just use ln(x) as the integrated power rather than x^alpha+1/alpha+1.
Just checking that all is well, then I'll do a pull request.
